### ISSUE TITLE
FlexibleTimeWindow.Mode: OFF incorrectly flagged as error E3030

### ISSUE THREAD

**Issue by na-memt** [aws-cloudformation/cfn-lint#4119]
### CloudFormation Lint Version

1.30.0

### What operating system are you using?

macOS

### Describe the bug

When using AWS::Scheduler::Schedule resources with FlexibleTimeWindow.Mode: OFF (without quotes), cfn-lint incorrectly reports error E3030 stating that "False is not one of ['OFF', 'FLEXIBLE']".

This appears to be a case where cfn-lint is incorrectly interpreting the uppercase string OFF as a boolean value False, despite YAML specifications indicating that uppercase OFF should be treated as a string.

### Expected behavior

cfn-lint should recognize OFF as a valid string value for FlexibleTimeWindow.Mode since it's a valid value according to the AWS Scheduler API documentation and is correctly deployed by CloudFormation.

### Reproduction template

```yaml
AWSTemplateFormatVersion: "2010-09-09"
Resources:
  WeekdayStartSchedule:
    Type: AWS::Scheduler::Schedule
    Properties:
      Name: my-schedule
      Description: Schedule example
      FlexibleTimeWindow:
        Mode: OFF    # This triggers error E3030
      ScheduleExpression: cron(0 0 ? * MON-FRI *)
      ScheduleExpressionTimezone: Asia/Tokyo
      State: ENABLED
      Target:
        Arn: !Sub arn:${AWS::Partition}:scheduler:::aws-sdk:ecs:updateService
        RoleArn: !Ref SchedulerExecutionRole
        Input: '{}'
```

When running cfn-lint on this template, it incorrectly reports:

```
E3030 False is not one of ['OFF', 'FLEXIBLE']
```

According to YAML specification, uppercase OFF should be treated as a string, not a boolean value. The AWS CloudFormation service itself correctly processes templates with Mode: OFF without quotes.

Workaround: Adding quotes (Mode: "OFF") resolves the issue but shouldn't be necessary according to YAML specifications.

**Comment by kddejong** [comment]
YAML 1.1 specs [definition](https://yaml.org/type/bool.html) of boolean and CloudFormation [docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/template-formats.html#template-formats.supported-specifications).  Do you have any links on where you are seeing YAML doesn't treat OFF as a boolean value?

---
Going to do some validating on how CloudFormation is handling this but also need to evaluate how this resource is handling it.

**Comment by kddejong** [comment]
```
Resources:
  Parameter:
    Type: AWS::SSM::Parameter
    Properties:
      Type: String
      Value: OFF
```

results in a SSM parameter with value `false`.  So OFF in this case is translated to the expected value.  Going to take a quick look at the specific handler to see if its doing any special handling.

**Comment by kddejong** [comment]
Also with early validation: `Properties validation failed for resource WeekdayStartSchedule with message: [#/FlexibleTimeWindow/Mode: false is not a valid enum value]`

My suggestion would be to see this to `"OFF"` in compliance with YAML and the schema enum.  

What region are you deploying in?  I'm curious as this error is from `us-east-1`? Additionally when was the last time you updated this resource (which would have triggered early validation)?

**Comment by na-memt** [comment]
I was encountering the same issue with AWS::Scheduler::Schedule resources. Changing the FlexibleTimeWindow Mode from "STRICT" to "OFF" fixed the deployment for me. 

Before:
```yaml
FlexibleTimeWindow:
  Mode: "STRICT"
```


Thanks!
